static/app/utils/useCleanQueryParamsOnRouteLeave.tsx

```diff
@@ -0,0 +1,66 @@
+import {useCallback, useEffect} from 'react';
+import {browserHistory} from 'react-router';
+import type {Location} from 'history';
+
+import {useLocation} from 'sentry/utils/useLocation';
+
+type Opts = {
+  fieldsToClean: string[];
+};
+
+export function handleRouteLeave<Q extends object>({
+  fieldsToClean,
+  newLocation,
+  oldPathname,
+}: {
+  fieldsToClean: string[];
+  newLocation: Location<Q>;
+  oldPathname: string;
+}) {
+  const hasSomeValues = fieldsToClean.some(
+    field => newLocation.query[field] !== undefined
+  );
+
+  if (
+    newLocation.pathname === oldPathname ||
+    (newLocation.pathname !== oldPathname && !hasSomeValues)
+  ) {
```

Comment:
You can simplify this to be: 
```suggestion
  if (newLocation.pathname === oldPathname || !hasSomeValues) {
```

Reply:
Simplification according to wolfram alpha: https://www.wolframalpha.com/input?key=&i=a+%7C%7C+%28%21a+%26%26+b%29